htslib/sam.h

```diff
@@ -2065,14 +2081,20 @@ Bit 2: Recalculate BAQ, even if a BQ tag is present.
  Force BAQ to be recalculated.  Note that a ZQ:Z tag will always disable
  recalculation.
 
+Bit 3 onwards: Choose parameters tailored to a specific instrument type.
+
+For ease of use these flags have pre-shifted enum values in htsRealnFlags.
+Hence these can simply be ORed together, such as in:
+    sam_prob_realn(b, ref, len, BAQ_APPLY | BAQ_EXTEND | BAQ_PACBIOCCS);
+
 @bug
 If the input read has both BQ:Z and ZQ:Z tags, the ZQ:Z one will be removed.
 Depending on what previous processing happened, this may or may not be the
 correct thing to do.  It would be wise to avoid this situation if possible.
 */
-
 HTSLIB_EXPORT
-int sam_prob_realn(bam1_t *b, const char *ref, hts_pos_t ref_len, int flag);
+int sam_prob_realn(bam1_t *b, const char *ref, hts_pos_t ref_len,
+                   enum htsRealnFlags flag);
```

Comment:
In theory this is an ABI breakage because the underlying type of an enumeration is implementation-defined and might not be the same size as `int`. In practice on modern platforms for parameter passing, there is probably not a problem.

Where this change certainly **does** cause a problem is in C++: any existing C++ code that calls `sam_prob_realn()` with an `int` for a flag will become a compile-time error.

Reply:
BTW ensuring the enum sizes are future-proofed even with `-fshort-enums` is why _hts.h_'s enums (`htsFormatCategory` and friends) have a rather goofy `foo_maximum = 32767` entry that's otherwise useless.

Reply:
Meh!  I did do a very cursory search which said it was int (but I suspect it was just comparable to int, rather than always represented as int). Anyway C++ as usual barges in to spoil the party!  I guess we'll just have to keep it as int and document the enum in comments then.

Thanks.